// office-ui-fabric-react/lib-amd/components/Layer/index.js
define(["require", "exports", "tslib", "./Layer", "./Layer.base", "./LayerHost", "./Layer.notification", "./Layer.styles"], function (require, exports, tslib_1, Layer_1, Layer_base_1, LayerHost_1, Layer_notification_1, Layer_styles_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    tslib_1.__exportStar(Layer_1, exports);
    tslib_1.__exportStar(Layer_base_1, exports);
    tslib_1.__exportStar(LayerHost_1, exports);
    exports.getLayerHostSelector = Layer_notification_1.getDefaultTarget;
    exports.getLayerCount = Layer_notification_1.getLayerCount;
    exports.getLayerHost = Layer_notification_1.getLayerHost;
    exports.notifyHostChanged = Layer_notification_1.notifyHostChanged;
    exports.registerLayer = Layer_notification_1.registerLayer;
    exports.registerLayerHost = Layer_notification_1.registerLayerHost;
    exports.setLayerHostSelector = Layer_notification_1.setDefaultTarget;
    exports.unregisterLayer = Layer_notification_1.unregisterLayer;
    exports.unregisterLayerHost = Layer_notification_1.unregisterLayerHost;
    exports.getLayerStyles = Layer_styles_1.getStyles;
});
