// powerbi-report-viewer/dist/pbireportviewer/reportViewer.styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.stackTokens1 = exports.stackTokens = exports.dropdownStyles = exports.controlClass = void 0;

var _officeUiFabricReact = require("office-ui-fabric-react");

var controlClass = (0, _officeUiFabricReact.mergeStyleSets)({
  filtercontainer: {},
  Embedcontainer: {
    height: '70vh',
    marginTop: "0px",
    width: "85vw"
  },
  searchStyles: {
    marginTop: "30px"
  },
  margintop5: {
    marginTop: "5px"
  },
  marginleft15: {
    marginLeft: "5px",
    textAlign: "left"
  },
  submitStyles: {
    marginTop: "28px",
    marginLeft: "10px"
  },
  optionsStyles: {
    marginTop: "0px",
    marginLeft: "30px",
    padding: "0px"
  },
  optionsStyles1: {
    width: '230px',
    marginTop: "0px",
    marginLeft: "0px",
    padding: "0px"
  },
  commandbarStyles: {
    padding: "0px",
    width: "85vw"
  }
});
exports.controlClass = controlClass;
var dropdownStyles = {
  dropdown: {
    width: 350,
    marginTop: "10px"
  }
};
exports.dropdownStyles = dropdownStyles;
var stackTokens = {
  childrenGap: 0
};
exports.stackTokens = stackTokens;
var stackTokens1 = {
  childrenGap: 5
};
exports.stackTokens1 = stackTokens1;
