// office-ui-fabric-react/lib-commonjs/components/Layer/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
tslib_1.__exportStar(require("./Layer"), exports);
tslib_1.__exportStar(require("./Layer.base"), exports);
tslib_1.__exportStar(require("./LayerHost"), exports);
var Layer_notification_1 = require("./Layer.notification");
exports.getLayerHostSelector = Layer_notification_1.getDefaultTarget;
exports.getLayerCount = Layer_notification_1.getLayerCount;
exports.getLayerHost = Layer_notification_1.getLayerHost;
exports.notifyHostChanged = Layer_notification_1.notifyHostChanged;
exports.registerLayer = Layer_notification_1.registerLayer;
exports.registerLayerHost = Layer_notification_1.registerLayerHost;
exports.setLayerHostSelector = Layer_notification_1.setDefaultTarget;
exports.unregisterLayer = Layer_notification_1.unregisterLayer;
exports.unregisterLayerHost = Layer_notification_1.unregisterLayerHost;
var Layer_styles_1 = require("./Layer.styles");
exports.getLayerStyles = Layer_styles_1.getStyles;
