// office-ui-fabric-react/lib-commonjs/Theme.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
//When adding new exports to this file, also add them to index.ts.
var theme_1 = require("@fluentui/theme");
exports.AnimationStyles = theme_1.AnimationStyles;
exports.AnimationVariables = theme_1.AnimationVariables;
exports.CommunicationColors = theme_1.CommunicationColors;
exports.DefaultEffects = theme_1.DefaultEffects;
exports.DefaultFontStyles = theme_1.DefaultFontStyles;
exports.DefaultPalette = theme_1.DefaultPalette;
exports.DefaultSpacing = theme_1.DefaultSpacing;
exports.Depths = theme_1.Depths;
exports.FluentTheme = theme_1.FluentTheme;
exports.createFontStyles = theme_1.createFontStyles;
exports.createTheme = theme_1.createTheme;
exports.LocalizedFontFamilies = theme_1.LocalizedFontFamilies;
exports.LocalizedFontNames = theme_1.LocalizedFontNames;
exports.mergeThemes = theme_1.mergeThemes;
exports.MotionDurations = theme_1.MotionDurations;
exports.MotionTimings = theme_1.MotionTimings;
exports.MotionAnimations = theme_1.MotionAnimations;
exports.NeutralColors = theme_1.NeutralColors;
exports.SharedColors = theme_1.SharedColors;
exports.registerDefaultFontFaces = theme_1.registerDefaultFontFaces;
tslib_1.__exportStar(require("@fluentui/react-theme-provider"), exports);
//When adding new exports to this file, also add them to index.ts.
